host/sgx/sgxload.c

```diff
@@ -364,8 +367,30 @@ oe_result_t oe_sgx_create_enclave(
 #endif // OEHOSTMR
     }
 
+    /*
+     * Load desired enclave start address. NOTE: Currently, this value is NULL
+     * when zero base enclave is not enabled. Also, base_address has to be a
+     * multiple of OE_SGX_PAGE_SIZE.
+     */
+    base_address = (void*)context->start_address;
+    if ((uint64_t)base_address & OE_SGX_PAGE_SIZE_MASK)
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
+    }
+
+    if (context->create_zero_base_enclave)
+    {
+        enclave_elrange.enclave_image_address = (uint64_t)base_address;
+        enclave_elrange.elrange_start_address = (uint64_t)OE_ADDRESS_ZERO;
+        enclave_elrange.elrange_size = enclave_size;
+
+        ex_features = ENCLAVE_CREATE_EX_EL_RANGE;
+        ex_features_array[ENCLAVE_CREATE_EX_EL_RANGE_BIT_IDX] =
+            &enclave_elrange;
+    }
+
     /* Create SECS structure */
-    if (!(secs = _new_secs((uint64_t)base, enclave_size, context)))
+    if (!(secs = _new_secs((uint64_t)OE_ADDRESS_ZERO, enclave_size, context)))
```

Comment:
This seems affect the logic when `oe_sgx_is_simulation_load_context` is true where the base will be set and then passed to _new_secs. Can you double check?

Reply:
Corrected the bug, wrt the simulation mode secs->base value.

As for the enclave_size, the elrange_size is the secs->size that is being recorded by the PSW code and no base alignments are done. https://github.com/intel/linux-sgx/blob/create_enclave_at_0/psw/enclave_common/sgx_enclave_common.cpp#L634-L638

And the secs->size is set to elrange_size - https://github.com/intel/linux-sgx/blob/1784a39bf80fe5bbfe653459b1bee0184c834e06/psw/enclave_common/sgx_enclave_common.cpp#L734